sources/PTYTab.m

```diff
@@ -1855,51 +1856,86 @@ - (NSSize)_minSizeOfView:(NSView*)view
 }
 
 // Blur the window if any session is blurred.
-- (bool)blur
-{
-    int n = 0;
-    int y = 0;
+- (BOOL)blur {
     NSArray* sessions = [self sessions];
     for (PTYSession* session in sessions) {
         if ([session transparency] > 0 &&
             [[session textview] useTransparency] &&
             [[[session profile] objectForKey:KEY_BLUR] boolValue]) {
-            ++y;
-        } else {
-            ++n;
+            return YES;
         }
     }
-    return y > 0;
+    return NO;
 }
 
-- (double)blurRadius
-{
+- (double)averageBlurRadiusForInactive:(BOOL)inactive {
     double sum = 0;
     double count = 0;
     NSArray* sessions = [self sessions];
     for (PTYSession* session in sessions) {
-        if ([[[session profile] objectForKey:KEY_BLUR] boolValue]) {
-            sum += [[session profile] objectForKey:KEY_BLUR_RADIUS] ? [[[session profile] objectForKey:KEY_BLUR_RADIUS] floatValue] : 2.0;
+        BOOL useBlur = inactive ? [session inactiveBlur] : [session blur];
+        if (useBlur) {
+            double blurRadius = inactive ? [session inactiveBlurRadius] : [session blurRadius];
+            sum += blurRadius;
             ++count;
         }
     }
     if (count > 0) {
         return sum / count;
     } else {
-        // This shouldn't actually happen, but better save than divide by zero.
+        // This shouldn't actually happen, but better safe than divide by zero.
         return 2.0;
     }
 }
 
-- (void)recheckBlur
-{
+- (double)blurRadius {
+    return [self averageBlurRadiusForInactive:NO];
+}
+
+- (BOOL)useInactiveTransparency {
+    NSArray *sessions = [self sessions];
+    for (PTYSession *session in sessions) {
+        if ([[session textview] useInactiveTransparency]) {
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (BOOL)inactiveBlur {
+    NSArray* sessions = [self sessions];
+    for (PTYSession* session in sessions) {
+        if ([session inactiveTransparency] > 0 &&
+            [session useInactiveTransparency] &&
+            [session inactiveBlur]) {
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (double)inactiveBlurRadius {
+    return [self averageBlurRadiusForInactive:YES];
+}
+
+- (void)recheckBlur {
     PtyLog(@"PTYTab recheckBlur");
     if ([realParentWindow_ currentTab] == self &&
         ![[realParentWindow_ window] isMiniaturized]) {
-        if ([self blur]) {
-            [parentWindow_ enableBlur:[self blurRadius]];
+        if ([[[self realParentWindow] window] isKeyWindow]){
```

Comment:
Does Xcode have a setting to make spacing consistent? I'm used to having reshaper do that for me in visual studio.


> On Mar 18, 2015, at 17:24, gnachman <notifications@github.com> wrote:
> 
> In sources/PTYTab.m:
> 
> >      PtyLog(@"PTYTab recheckBlur");
> >      if ([realParentWindow_ currentTab] == self &&
> >          ![[realParentWindow_ window] isMiniaturized]) {
> > -        if ([self blur]) {
> > -            [parentWindow_ enableBlur:[self blurRadius]];
> > +        if ([[[self realParentWindow] window] isKeyWindow]){
> Needs a space between ) and {
> 
> —
> Reply to this email directly or view it on GitHub.
>